src/tests/unit/test_tree_functionality/tree_dict_conversion.py

```diff
@@ -0,0 +1,328 @@
+import unittest
```

Comment:
the tests in this files are copied from the old "test_tree_node_helpers.py" file, without being modified. The only difference is that a new test "test_tree_node_to_ref_dict_2" has been added.